scripts/fast-reboot

```diff
@@ -121,12 +121,19 @@ function wait_for_pre_shutdown_complete_or_fail()
     waitcount=0
     # Wait up to 60 seconds for pre-shutdown to complete
     while [[ ${waitcount} -lt 600 ]]; do
-        STATE=`timeout 10 /usr/bin/redis-cli -n 6 hget "${TABLE}" state`
-        if [[ x"${STATE}" != x"requesting" ]]; then
+        # timeout doesn't work with -i option of "docker exec". Therefore we have
+        # to invoke docker exec directly below.
+        STATE=`timeout 5s docker exec database redis-cli -n 6 hget "${TABLE}" state; if [[ $? == 124 ]]; then echo "timed out"; fi`
+
+        if [[ x"${STATE}" == x"timed out" ]]; then
+            waitcount+=50
+            debug "Timed out getting pre-shutdown state (${waitcount}) ..."
+        elif [[ x"${STATE}" != x"requesting" ]]; then
             break
+        else
```

Comment:
I don't know how often we see the timeout or we see some significant time spent for CLI checking DB based on your experience.   In case the database check CLI took somewhere below 5seconds but much higher than 0.1 seconds,  this loop could end up getting much higher than 60seconds.

Reply:
Just one example so it is more clear:  If the redis-cli took some significant time say 1second.  but we got the STATE = "requesting" all the time, we are going to wait for 600 seconds in this loop.

Reply:
I was adding 50 to the loop counter when timeout happens to compensate for the 5 seconds timeout. :-)

Reply:
The normal redis-cli operations takes very little time to complete, with 600 retries, it is still roughly 60 seconds.